src/webui/webapplication.cpp

```diff
@@ -125,14 +125,18 @@ namespace
     QString createLanguagesOptionsHtml()
     {
         // List language files
-        const QDir langDir {u":/www/translations"_s};
-        const QStringList langFiles = langDir.entryList(QStringList(u"webui_*.qm"_s), QDir::Files);
+        const QStringList langFiles = QDir(u":/www/translations"_s)
+            .entryList({u"webui_*.qm"_s}, QDir::Files, QDir::Name);
+
         QStringList languages;
+        languages.reserve(langFiles.size());
+
         for (const QString &langFile : langFiles)
         {
-            const QString localeStr = langFile.section(u"_"_s, 1, -1).section(u"."_s, 0, 0); // remove "webui_" and ".qm"
-            languages << u"<option value=\"%1\">%2</option>"_s.arg(localeStr, Utils::Misc::languageToLocalizedString(localeStr));
-            qDebug() << "Supported locale:" << localeStr;
+            const QString langCode = QStringView(langFile).sliced(6).chopped(3).toString(); // remove "webui_" and ".qm"
```

Comment:
It could be optimized even more, because `langCode` also doesn't have to be a `QString`. However, to do this, you need to change `languageToLocalizedString()` to accept `QStringView`.
If you want.

Reply:
Done.